sklearn/feature_selection/_from_model.py

```diff
@@ -494,6 +494,7 @@ def get_metadata_routing(self):
         router = MetadataRouter(owner=self.__class__.__name__).add(
             estimator=self.estimator,
             method_mapping=MethodMapping()
+            .add(caller="fit_transform", callee="fit_transform")
```

Comment:
```suggestion
            .add(caller="fit_transform", callee="fit")
```

On the top of the head, I would expect `callee` to be `fit` because this is the step that requires metadata for this specific selector.

However, I'm always confused with the routing/caller/callee :).

@StefanieSenger @adrinjalali would you mind to have a look.

Reply:
we shouldn't have to add these _composite_ methods here, they should be auto generated from the simple methods. This might be an issue in the `_metadata_requests.py` that I need to debug to see where the issue is.

Reply:
I was also curious about @glemaitre's question. It actually doesn't matter whether you use `callee="fit_transform"` or `callee="fit"` for the fix. Both values will result in successful metadata routing. 

I _think_ this is because the Feature Selector's `fit_transform` function passes the `**fit_params` to `self.fit()` without explicitly using the metadata routing functionality. Code snippet below ([source](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/base.py#L863)) shows that its only using the metadata routing for validation and has no impact on the `**fit_params`. 
```python
    def fit_transform(self, X, y=None, **fit_params):
        if _routing_enabled():
            transform_params = self.get_metadata_routing().consumes(
                method="transform", params=fit_params.keys()
            )
            if transform_params:
                warnings.warn(
                    (
                        f"This object ({self.__class__.__name__}) has a `transform`"
                        " method which consumes metadata, but `fit_transform` does not"
                        " forward metadata to `transform`. Please implement a custom"
                        " `fit_transform` method to forward metadata to `transform` as"
                        " well. Alternatively, you can explicitly do"
                        " `set_transform_request`and set all values to `False` to"
                        " disable metadata routed to `transform`, if that's an option."
                    ),
                    UserWarning,
                )
        if y is None:
            # fit method of arity 1 (unsupervised transformation)
            return self.fit(X, **fit_params).transform(X)
        else:
            # fit method of arity 2 (supervised transformation)
            return self.fit(X, y, **fit_params).transform(X)
```

To illustrate the function calls, in order:

1. `pipeline.fit()`:
    - First calls `process_routing(_method='fit')` to request the metadata to pass forward
    - Then calls `feature_selector.fit_transform()` and includes the routed metadata
2. `feature_selector.fit_transform()`:
    - does NOT call `process_routing()` 
    - calls `self.fit(X, y, **fit_params)`
3. `feature_selector.fit()`:
    - First calls `process_routing(_method='fit')` to request the metadata to pass forward
    - Then calls `estimator.fit()` and includes the routed metadata

Because step 2 doesn't participate in the metadata routing, it doesn't matter what the `callee` value is for the `feature_selector`. It just has to be one of the generically valid values defined by the class (e.g., `fit`, `transform`, `fit_transform`, etc).